From 6bf279b6c1f868f2e6df9567c664731037d9646c Mon Sep 17 00:00:00 2001
From: Mike Gorse <mgorse@suse.com>
Date: Thu, 10 Sep 2026 13:53:03 -0500
Subject: [PATCH] plugins: Ensure that SGI tables are initialized

Call g_try_malloc0, rather than g_try_malloc, when allocating tables.
This avoids potentially using uninitialized memory and matches the previous
behavior of calling calloc.

Related to #16494
---
 plug-ins/file-sgi/sgi-lib.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/plug-ins/file-sgi/sgi-lib.c b/plug-ins/file-sgi/sgi-lib.c
index 534ceaaa25..2c6e9f8f4e 100644
--- a/plug-ins/file-sgi/sgi-lib.c
+++ b/plug-ins/file-sgi/sgi-lib.c
@@ -352,8 +352,8 @@ sgiOpenFile(FILE *file, /* I - File to open */
                 free(sgip);
                 return (NULL);
               }
-            sgip->table[0] = g_try_malloc ((gsize) sgip->ysize * sgip->zsize *
-                                           sizeof (goffset));
+            sgip->table[0] = g_try_malloc0 ((gsize) sgip->ysize * sgip->zsize *
+                                            sizeof (goffset));
             if (sgip->table[0] == NULL)
               {
                 free(sgip->table);
@@ -454,8 +454,8 @@ sgiOpenFile(FILE *file, /* I - File to open */
                   return (NULL);
                 }
               sgip->table[0] =
-                g_try_malloc ((gsize) sgip->ysize * sgip->zsize *
-                              sizeof (goffset));
+                g_try_malloc0 ((gsize) sgip->ysize * sgip->zsize *
+                               sizeof (goffset));
               if (sgip->table[0] == NULL)
                 {
                   free(sgip->table);
@@ -467,8 +467,8 @@ sgiOpenFile(FILE *file, /* I - File to open */
                 sgip->table[i] = sgip->table[0] + i * sgip->ysize;
               sgip->length    = calloc(sgip->zsize, sizeof(goffset *));
               sgip->length[0] =
-                g_try_malloc ((gsize) sgip->ysize * sgip->zsize *
-                              sizeof (goffset));
+                g_try_malloc0 ((gsize) sgip->ysize * sgip->zsize *
+                               sizeof (goffset));
               for (i = 1; i < sgip->zsize; i ++)
                 sgip->length[i] = sgip->length[0] + i * sgip->ysize;
               break;
-- 
2.55.0

